### PR TITLE
ENT-3522: Request for field to specify locale in Django enterprise accounts

### PR DESCRIPTION
**Pull request by saleem-latif**
**Jira Ticket:** [ENT-3522](https://openedx.atlassian.net/browse/ENT-3522)

__Description:__
1. Add a new column in enterprise customer config model, that will be used by the customers to change the learner's default language
2. Introduce a new middleware that will check If a linked enterprise learner’s enterprise has a non-null preferred language, then that learner will have their language cookie set to the preferred language (assuming they themselves have not set a language preference on the account settings page).
3.  There is already a middleware in `edx-platform/lang_pref` that can be used as a reference for the implementation https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/lang_pref/middleware.py#L52
4. All experiences on edX influenced by the edX language toggle  “English/Spanish” will be updated accordingly. 

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [x] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/middleware.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        if hasattr(request, 'user'):
+            current_user = getattr(request.user, 'real_user', request.user)
+
+        if current_user and current_user.is_authenticated:
+            enterprise_customer = get_enterprise_customer_for_user(current_user)
+
+            if not (enterprise_customer and enterprise_customer.default_language):
+                # If user does not belong to an enterprise customer or the default language for the user's
+                # enterprise customer is not set, then no need to go any further.
+                return response
```
Comment (sameenfatima78) [inline enterprise/middleware.py:64]: Are we handling the case when enterprise default language is not set but user's language preference exists. Wouldn't the expected behaviour in that case be to go with user's preference? I'm guessing this scenario is being handled in edx-platform's middleware that will run after this middleware, correct?

**Review by sameenfatima78 — APPROVED** [review]
Looks good... just one question

File: enterprise/middleware.py (hunk 1 above)
Comment (saleem-latif) [reply enterprise/middleware.py:64]: Yes, that is correct. That behavior is being handled by the middleware in the edx-platform.

File: enterprise/middleware.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            # Get the user's language preference
+            try:
+                user_pref = get_user_preference(current_user, LANGUAGE_KEY)
+            except (UserAPIRequestError, UserAPIInternalError):
+                # Ignore errors related to user preferences not found.
+                pass
+
+            # If user's language preference is not set and enterprise customer has a default language configured
+            # then set the default language as the learner's language
+            if not user_pref and not is_request_from_mobile_app(request) and enterprise_customer.default_language:
```
Comment (hasnain-naveed) [inline enterprise/middleware.py:75]: why are we checking `enterprise_customer.default_language`  here again as you have a strict if condition above for this case? Am I missing something?

**Review by hasnain-naveed — COMMENTED** [review]
One question why are we by-passing the mobile requests? I have seen the edx-platform middleware. Any idea just curious?

**Comment by hasnain-naveed** [comment]
One more thing, according to your story, I guess you should only provide an English or Spanish option in the admin panel? This is more of a business decision. Just letting you know.

File: enterprise/middleware.py (hunk 2 above)
Comment (saleem-latif) [reply enterprise/middleware.py]: Done

**Review by hasnain-naveed — APPROVED** [review]

**Review by zamanafzal — APPROVED** [review]
💯

**Review by muhammad-ammar — APPROVED** [review]
LGTM 👍

**Pull request merged**